logger.go

```diff
@@ -2,25 +2,51 @@ package svc
 
 import (
 	"os"
+	"strings"
 	"time"
 
+	"github.com/blendle/zapdriver"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
 func newLogger(name, version string, level zapcore.Level) (*zap.Logger, zap.AtomicLevel) {
+	logFormat := strings.ToLower(os.Getenv("LOG_FORMAT"))
+
 	atom := zap.NewAtomicLevel()
 	atom.SetLevel(level)
 
+	// stackdriver
+	if logFormat == "stackdriver" {
```

Comment:
I'm not a fan of having an hardcoded list of log formats we support.

Alternative suggestion: we already have a way to configure different kind of loggers through the options mechanism (`WithDevelopmentLogger()` and `WithProductionLogger()`). What about adding a third logger-related option giving more control to the user as to how the logger is configured? Something like:

```
func WithLogger(logger *zap.Logger, atom zap.AtomicLevel) Option {
	return func(s *SVC) error {
		return assignLogger(s, logger, atom)
	}
}
```

In addition to this "custom logger injector" we could also maybe provide utility options to use a stackdriver/console compatible ones:

```
func WithConsoleLogger(atom zap.AtomicLevel) Option {
	return func(s *SVC) error {
		config := zap.NewProductionEncoderConfig()
		config.EncodeTime = zapcore.RFC3339TimeEncoder
		enc := zapcore.NewConsoleEncoder(config)

		logger := zap.New(zapcore.NewSampler(zapcore.NewCore(
			enc,
			zapcore.Lock(os.Stdout),
			atom,
		), time.Second, 100, 10),
			zap.ErrorOutput(zapcore.Lock(os.Stderr)),
			zap.AddCaller(),
			zap.Fields(zap.String("app", s.Name), zap.String("version", s.Version)),
		)
		return assignLogger(s, logger, atom)
	}
}
```

Or maybe we should just provide options to modify the encoder used by the logger? (this would ensure that for console/stackdriver loggers we have the same sampling and default fields configuration)

The important thing is that the solution we choose should rely on the Options mechanism we have (no hardcoded list of supported loggers/encoders and no dependency on hidden environment variables)

Reply:
Excellent comments thanks. I totally missed that you had implemented the options pattern. Done!